### PR TITLE
added functionality for user, shopping cart and orders

### PR DESCRIPTION
**Pull request by mikhail771**

### CODE REVIEW

**Review by matviichuk92 — APPROVED** [review]
great job!

File: src/main/java/com/internet/mixshop/service/impl/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import com.internet.mixshop.model.ShoppingCart;
+import com.internet.mixshop.service.OrderService;
+import com.internet.mixshop.service.ShoppingCartService;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    OrderDao orderDao;
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/service/impl/OrderServiceImpl.java]: what about access modifier?

File: src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import com.internet.mixshop.lib.Service;
+import com.internet.mixshop.model.Product;
+import com.internet.mixshop.model.ShoppingCart;
+import com.internet.mixshop.service.ShoppingCartService;
+import java.util.Optional;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    ShoppingCartDao shoppingCartDao;
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java]: same

File: src/main/java/com/internet/mixshop/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        shoppingCartService.clear(shoppingCart);
+        return order;
+    }
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        return orderDao.getAllOrders()
+                .stream()
+                .filter(order -> order.getUserid().equals(userId))
+                .collect(Collectors.toList());
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/service/impl/OrderServiceImpl.java]: be more attentive 
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_14_2.html

File: src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(product);
+        return shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        ShoppingCart sc = shoppingCartDao.getById(shoppingCart.getId()).get();
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java]: sc - is not informative naming

File: src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        sc.getProducts().remove(product);
+        shoppingCartDao.update(sc);
+        return true;
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        if (shoppingCart.getProducts().remove(product)) {
            shoppingCartDao.update(shoppingCart);
            return true;
        }
        return false;
```

File: src/main/java/com/internet/mixshop/service/ShoppingCartService.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+public interface ShoppingCartService {
+    ShoppingCart create(ShoppingCart shoppingCart);
+
+    ShoppingCart addProduct(ShoppingCart shoppingCart, Product product);
+
+    boolean deleteProduct(ShoppingCart shoppingCart, Product product);
+
+    void clear(ShoppingCart shoppingCart);
+
+    Optional<ShoppingCart> getByUserId(Long userId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/service/ShoppingCartService.java]: wrong method declaration, check task description once again
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/homework/lesson_14_hw2.html

File: src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.getProducts().clear();
+    }
+
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long userId) {
+        return Storage.shoppingCarts.stream()
+                .filter(shoppingCart -> shoppingCart.getUserId().equals(userId))
+                .findFirst();
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java]: this logic should be implemented on dao layer

File: src/main/java/com/internet/mixshop/service/impl/UserServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    public User update(User user) {
+        return userDao.update(user);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        ShoppingCartService shopCartService = new ShoppingCartServiceImpl();
+        shopCartService.delete(shopCartService.getByUserId(id).get());
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/service/impl/UserServiceImpl.java]: we shouldn't delete shoppingCart when we delete user by default, in future we may have users without shoppingCart

File: src/main/java/com/internet/mixshop/service/impl/OrderServiceImpl.java (hunk 1 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/service/impl/OrderServiceImpl.java]: done

File: src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java (hunk 2 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java]: done

File: src/main/java/com/internet/mixshop/service/impl/OrderServiceImpl.java (hunk 3 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/service/impl/OrderServiceImpl.java]: done

File: src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java (hunk 4 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java]: done

File: src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java (hunk 5 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java]: done

File: src/main/java/com/internet/mixshop/service/ShoppingCartService.java (hunk 6 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/service/ShoppingCartService.java]: done

File: src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java (hunk 7 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/service/impl/ShoppingCartServiceImpl.java]: done

File: src/main/java/com/internet/mixshop/service/impl/UserServiceImpl.java (hunk 8 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/service/impl/UserServiceImpl.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**